Compile error during Go interface build

### Describe the bug

Compile fails with unrecognized name from C FFI

### Steps to Reproduce

### Setup and versions

git master @ 68fa8ee661deafc826716a72be88d629e5f41f38

### Additional information (depending on the issue)

```
Making all in bindings/go
make[2]: Entering directory '/lustre/projects/ucx-group/arcurtis/src/ucx/build/bindings/go'
cd /lustre/projects/ucx-group/arcurtis/src/ucx/build/../ucx-git/bindings/go/src/ucx ;\
go build
# ucx
./callbacks.go:92:26: could not determine kind of name for C.UCP_AM_RECV_ATTR_FIELD_REPLY_EP
./callbacks.go:88:50: could not determine kind of name for C.ucp_am_recv_param_t
cgo:
gcc errors for preamble:
In file included from ./callbacks.go:8:
./goucx.h:16:69: error: unknown type name 'ucp_am_recv_param_t'; did you mean 'ucp_mem_map_params_t'?
                                          void *data, size_t length, ucp_am_recv_param_t *param);
                                                                     ^~~~~~~~~~~~~~~~~~~
                                                                     ucp_mem_map_params_t

make[2]: *** [Makefile:614: build] Error 2
```

--- a/bindings/go/Makefile.am
+++ b/bindings/go/Makefile.am
@@ -7,7 +7,7 @@ if HAVE_GO
 
 GOOBJDIR=$(abs_top_builddir)/bindings/go/$(objdir)
 GOPATH=$(abs_top_srcdir)/bindings/go/
-CGOCFLAGS=-I$(abs_top_builddir)/src -I$(top_srcdir)/src
+CGOCFLAGS=-I$(abs_top_builddir)/src -I$(abs_top_srcdir)/src
 CGOLDFLAGS=-L$(abs_top_builddir)/src/ucp/$(objdir) -lucp -L$(abs_top_builddir)/src/ucs/$(objdir) -lucs
 UCX_SOPATH=$(abs_top_builddir)/src/ucp/$(objdir):$(abs_top_builddir)/src/ucs/$(objdir):$(abs_top_builddir)/src/ucm/$(objdir):$(abs_top_builddir)/src/uct/$(objdir)
 GOTMPDIR=$(GOOBJDIR)/tmp

